test/VaultAsyncMode.spec.ts

```diff
@@ -0,0 +1,335 @@
+
+    const ustDeployment = await deployments.get('UST');
+
+    underlying = MockUST__factory.connect(ustDeployment.address, owner);
+  });
+
+  beforeEach(() => fixtures());
+
+  beforeEach(async () => {
+    [owner, alice, bob] = await ethers.getSigners();
+
+    let Vault = await ethers.getContractFactory('Vault');
+    let MockStrategy = await ethers.getContractFactory('MockStrategyAsync');
+
+    vault = await Vault.deploy(
+      underlying.address,
+      TWO_WEEKS,
+      INVEST_PCT,
+      TREASURY,
+      owner.address,
+      PERFORMANCE_FEE_PCT,
+      INVESTMENT_FEE_PCT,
+      [],
+    );
+
+    underlying.connect(owner).approve(vault.address, MaxUint256);
+    underlying.connect(alice).approve(vault.address, MaxUint256);
+    underlying.connect(bob).approve(vault.address, MaxUint256);
+
+    strategy = await MockStrategy.deploy(vault.address, underlying.address);
+
+    await vault.connect(owner).setStrategy(strategy.address);
+
+    ({ addUnderlyingBalance, addYieldToVault } = createVaultHelpers({
+      vault,
+      underlying,
+    }));
+
+    await addUnderlyingBalance(alice, '1000');
+    await addUnderlyingBalance(bob, '1000');
+  });
+
+  describe("#claimYield doesn't rebalance the Vault's funds", () => {
+    it('reverts when yield > reserves', async () => {
+      const params = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(alice.address).build()],
+      });
+      await vault.connect(alice).deposit(params);
+
+      await addYieldToVault('50');
+      await vault.connect(owner).updateInvested();
+
+      await expect(
+        vault.connect(alice).claimYield(alice.address),
+      ).to.be.revertedWith('VaultNotEnoughFunds');
+    });
+
+    it("doesn't revert and withdraws funds from the Vault when yield = reserves", async () => {
+      const params = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(alice.address).build()],
+      });
+      await vault.connect(alice).deposit(params);
+
+      await addYieldToVault('50');
+      await vault.connect(owner).updateInvested();
+
+      // bob's deposit is exactly enough to cover alice's claim and leave valut with 0 balance
+      const bobsDeposit = depositParams.build({
+        amount: parseUnits('34'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(bob.address).build()],
+      });
+      await vault.connect(bob).deposit(bobsDeposit);
+
+      await expect(vault.connect(alice).claimYield(alice.address)).to.not.be
+        .reverted;
+
+      expect(await underlying.balanceOf(vault.address)).to.eq('0');
+      expect(await underlying.balanceOf(strategy.address)).to.eq(
+        parseUnits('135'),
+      );
+    });
+
+    it("doesn't revert and withdraws funds from the Vault when yield < reserves", async () => {
```

Comment:
I don't think we need "doesn't revert" section

Reply:
I tried to be more specific but I agree it is not necessary

Reply:
[fixed](https://github.com/lindy-labs/sc_solidity-contracts/pull/136/commits/47be529313b023a0fe04a06247a0095ceba141c9)